purchase_request/models/stock_rule.py

```diff
@@ -124,4 +126,22 @@ def create_purchase_request(self, procurement_group):
                 pr.write({"origin": procurement.origin})
         # Create Line
         request_line_data = rule._prepare_purchase_request_line(pr, procurement)
+        # check if request has lines for same product and data
+        # if yes, update qty instead of creating new line
+        same_product_date_request_line = pr.line_ids.search(
```

Comment:
WARNING: Model::search is a model method not a recordset method. if you want to filter a recordset with a restriction, use Model::filtered_domain
```suggestion
        same_product_date_request_line = pr.line_ids.filtered_domain(
```